### DISCUSSION TITLE
RFC 9113 in nghttp2 v1.49.0

### DISCUSSION THREAD

**Discussion by Keitagit-kun** [curl/curl#9386] [Q&A] [answered]
updated libnghttp2 to v1.49.0 and get errors `HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)` on many sites.  Then found [curl not functioning properly after updating to libnghttp2 1.49.0](https://github.com/nghttp2/nghttp2/issues/1786). 
Is it possible from curl side (easy/multy) to make some workaround to this issue? Or downgrade it to warning so sites would be still accessible?

**Comment by bagder** [comment]
> Is it possible from curl side (easy/multy) to make some workaround to this issue

Sure, as @tatsuhiro-t says in the nghttp2 issue: it can set a nghttp2 callback that can then work around it.

> Or downgrade it to warning so sites would be still accessible?

I believe that would still need to use the same approach but I don't know of nghttp2 provides any other mechanism.

**Comment by bagder** [comment, marked as answer]
nghttp2 1.50.0 will feature an option that relaxes the check.

curl in master (and the future 7.86.0 release) sets that new option if a new enough nghttp2 is found when building curl.
